### DISCUSSION TITLE
Z-Axis homes after first layer has been coloured

### DISCUSSION THREAD

**Discussion by dylan4397** [Sakati84/3DPrintColorizer#42] [Q&A]
Hey everybody, for the past two or three days I have been trying to get the colourizer working. 

The problem I have been having is the z-axis homing/coming too low after colouring the first layer. The printer will print/colour the first layer perfectly fine. Once it is ready to do the second layer it comes down too low and ruins the first layer. I have tried many ways to fix this, but I cannot figure it out. I thought that this may only be affecting the second layer, but I let it continue and the z-axis never rises. 

I set up the colourizer by following the directions (I went through them many times). I am using the stock Cura profile included with the download. 

My printer is a fairly stock ender 3 pro.

Here is a video of the problem. You can see that the printer successfully colours the first layer, but crashes into it when printing the second layer.

https://www.youtube.com/watch?v=rE8D3HNFsEk

Note: I have done normal prints (no sharpies) since encountering this problem and they have not had it. 

Thank you for the help! :)

**Comment by Sakati84** [comment]
Hi, can you post the G-Code? This would help figuring out if it´s a software problem or your printer missing some steps on the Z-Axis. What is your Z speed?

**Reply by dylan4397** [reply]
Thanks for the reply! My Z max is set at 25.
Here is the G-code. 
[SquareTest.gcode.zip](https://github.com/Sakati84/3DPrintColorizer/files/6992699/SquareTest.gcode.zip)

**Reply by Sakati84** [reply]
The G-Code looks absolutely fine to me. The second layer prints at a height of 0.4 mm - as it should. Perhaps your Z-Axis is missing some steps. Can you try it with a lower Z-Speed and/or only using the first pen?

**Reply by dylan4397** [reply]
I tried with a z speed of 8, but that made no difference with all of the pens. When I only used the first pen and a speed of 8, it printed and coloured the first 3 layers fine. When it got to layer 4, it starting coming too low again.
